python/lsst/daf/butler/script/queryDatasets.py

```diff
@@ -214,12 +205,26 @@ def getTables(self) -> list[AstropyTable]:
 
         return [table.getAstropyTable(datasetTypeName) for datasetTypeName, table in tables.items()]
 
-    def getDatasets(self) -> DatasetQueryResults:
-        """Get the datasets as a list of ``DatasetQueryResults``.
+    def getDatasets(self) -> Iterator[DatasetRef]:
+        """Get the datasets as a list.
 
         Returns
         -------
-        refs : `lsst.daf.butler.registry.queries.DatasetQueryResults`
+        refs : `list`
```

Comment:
Now it's an iterator.

Reply:
Yes. Thanks. In my first pass it was a list and then I had to reorganize because of the prune-datasets command line mocking.